x-pack/legacy/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts

```diff
@@ -267,32 +239,30 @@ describe('onPostAuthRequestInterceptor', () => {
         },
       ];
 
-      const response = await request('/s/not-found/app/kibana', spaces);
+      const { response } = await request('/s/not-found/app/kibana', spaces);
 
-      expect(response.statusCode).toEqual(302);
-      expect(response.headers.location).toEqual(serverBasePath);
+      expect(response.status).toEqual(302);
+      expect(response.header.location).toEqual(`/spaces/space_selector`);
     }, 30000);
-  });
 
-  it('when accessing the kibana app it always allows the request to continue', async () => {
-    const spaces = [
-      {
-        id: 'a-space',
-        type: 'space',
-        attributes: {
-          name: 'a space',
-          disabledFeatures: ['feature-1', 'feature-2', 'feature-4', 'feature-5'],
+    it('when accessing the kibana app it always allows the request to continue', async () => {
+      const spaces = [
+        {
+          id: 'a-space',
+          type: 'space',
+          attributes: {
+            name: 'a space',
+            disabledFeatures: ['feature-1', 'feature-2', 'feature-4', 'feature-5'],
+          },
         },
-      },
-    ];
+      ];
 
-    const response = await request('/s/a-space/app/kibana', spaces);
+      const { response } = await request('/s/a-space/app/kibana', spaces);
 
-    expect(response.statusCode).toEqual(200);
-  }, 30000);
+      expect(response.status).toEqual(200);
+    }, 30000);
 
-  describe('when accessing an API endpoint within a non-existent space', () => {
-    it('allows the request to continue', async () => {
+    it('allows the request to continue when accessing an API endpoint within a non-existent space', async () => {
```

Comment:
This would allow users to, for example, populate a non-existent space with saved objects (via the `saved_objects` API, or the `import` API), and then when the space is eventually created, it would already contain these objects.

It's not a feature that I would consider useful, but it was done this way for a couple of reasons:
1) Kibana has historically not enforced these types of integrity constraints in the past
2) It was easier from an implementation perspective when it comes to securing access

Reply:
Aha! I see now, thanks!